gateway/src/cluster/builder.rs

```diff
@@ -90,12 +97,29 @@ impl ClusterBuilder {
             queue: self.queue,
             resume_sessions: self.resume_sessions,
             shard_presence: self.shard_presence,
-            shard_scheme: self.shard_scheme,
+            shard_scheme: self.shard_scheme.expect("always set"),
         };
 
         Cluster::new_with_config(config, shard_config).await
     }
 
+    async fn retrieve_connect_info(http: &Client) -> Result<BotConnectionInfo, ClusterStartError> {
```

Comment:
this could be documented

Reply:
I feel like there's not much more to say other than "retrieves [`BotConnectionInfo`]", but that's obvious imo, and thus not worth it